docs/developer/decision-records/2022-07-28-transfer-process-new-state/README.md

```diff
@@ -0,0 +1,31 @@
+# TransferProcess new "Provisioning Requested" state
+
+## Decision
+
+Since the provisioning phase became "async" with the introduction of the `http-provisioner` extension,
+there's the need to introduce a new state that describes a transfer process for which the provisioning has been requested
+but it hasn't been completed yet.
+
+## Rationale
+
+The classic provision method (see `S3BucketProvisioner` or `ObjectStorageProvisioner`), despite providing `async` interface,
+works in a synchronous way: when the `CompletableFuture` returned by the provisioner completes, this means that the actual
+provisioning process completed and the `StateMachineManager` can fetch that `TransferProcess` and continue its processing.
+The `http-provisioner` works in a different way: when the `CompletableFuture` completes, this means that the provisioning
+has been requested, but it will be completed in the future. At that point the `TransferProcess` should stay in a state
+that would make the `StateMachineManager` ignore it, as it would be up to the external provisioner to complete the provisioning
+calling back the connector.
+
+The same rationale goes for the `deprovisioning` phase.
+
+## Approach
+
+As we already did for the execution of "asynchronous operations" like contract negotiation, the solution would be to add
+a new state that is not processed by the `StateMachineManager`, in this case the choice would be to add:
+- `PROVISIONING_REQUESTED(250)` (between `PROVISIONING(200)` and `PROVISIONED(300)`)
```

Comment:
don't want to stir the pot too much, but one could also argue that the state sequence goes `PROVISIONING_REQUESTED` (request sent to provisioner) -> `PROVISIONING` (while the provisioner is working) -> `PROVISIONED`, no?

Reply:
makes sense, done